src_python/habitat_sim/utils/datasets_download.py

```diff
@@ -697,33 +697,35 @@ def main(args):
     args = parser.parse_args(args)
     replace = args.replace
 
-    # get a default data_path "./data/"
-    data_path = args.data_path
-    if not data_path:
-        try:
-            data_path = os.path.abspath("./data/")
-            print(
-                f"No data-path provided, default to: {data_path}. Use '--data-path' to specify another location."
-            )
-            if not os.path.exists(data_path):
-                os.makedirs(data_path)
-        except Exception:
-            traceback.print_exc(file=sys.stdout)
-            print("----------------------------------------------------------------")
-            print(
-                "Aborting download, failed to create default data_path and none provided."
-            )
-            print("Try providing --data-path (e.g. '/path/to/habitat-sim/data/')")
-            print("----------------------------------------------------------------")
-            parser.print_help()
-            exit(2)
+    default_data_path = "./data"
+    data_path = os.path.realpath(
+        args.data_path if args.data_path else default_data_path
+    )
+    if not args.data_path:
+        print(
+            f"No data-path provided, defaults to: {default_data_path}. Use '--data-path' to specify another location."
+        )
+        if os.path.islink(default_data_path):
+            print(f"Note, {default_data_path} is a symbolic link that points to {data_path}.")
+
+    try:
+        os.makedirs(data_path, exist_ok=True)
```

Comment:
What kind of errors can arise here? I think it would just be permission errors? 
What is the advantage of printing these messages over just raising the error?

Reply:
To be honest, I didn't ask myself this questions during fixing, but it seems to me that this is a "graceful way" to exit a program. Exception should be raised on program errors. Here everything is fine, user specified wrong path and we inform him about it. Why the program should fail with exception?

Reply:
@rpartsey probably should still be restricted to IOException? LIke no reason to catch SystemError exceptions from Python etc.